kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -281,18 +291,25 @@ private CloseableIterator<FilteredColumnarBatch> applyDataSkipping(
                 Arrays.asList(dataSkippingFilter, Literal.ofBoolean(true))),
             AlwaysTrue.ALWAYS_TRUE);
 
-        PredicateEvaluator predicateEvaluator = engine
-            .getExpressionHandler()
-            .getPredicateEvaluator(prunedStatsSchema, filterToEval);
+        PredicateEvaluator predicateEvaluator = wrapEngineException(
+            () -> engine
+                .getExpressionHandler()
+                .getPredicateEvaluator(prunedStatsSchema, filterToEval),
+            "Get the predicate evaluator for data skipping with schema=%s and filter=%s",
+            prunedStatsSchema,
+            filterToEval
+        );
```

Comment:
can we create something like this? Separates out the context from the actual task.

```
exceptionWrapper()
   .withContext(msg, args)
   .wrap(() -> task)
```

Reply:
In theory it seems cleaner but I tried it out in the code and honestly not sure it's any more readable to me
```java
                    predicateEvaluator = exceptionWrapper()
                        .withContext(
                            "Get the predicate evaluator for partition pruning with schema=%s and" +
                                " filter=%s", next.getData().getSchema(), predicateOnScanFileBatch)
                        .wrap(() -> engine.getExpressionHandler().getPredicateEvaluator(
                                next.getData().getSchema(),
                                predicateOnScanFileBatch)
                        );

                    predicateEvaluator = wrapEngineException(
                        () -> engine.getExpressionHandler().getPredicateEvaluator(
                            next.getData().getSchema(),
                            predicateOnScanFileBatch),
                        "Get the predicate evaluator for partition pruning with schema=%s and" +
                            " filter=%s",
                        next.getData().getSchema(),
                        predicateOnScanFileBatch
                    );
```